utilities/hydrofabric_test/test_ngen.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+
+from pathlib import Path
+from os import chdir
+import yaml
+from tempfile import mkdtemp
+import geopandas as gpd
+import pandas as pd
+from config import General
+from typing import Union
+import subprocess
+
+from ngen.config.formulation import Formulation
+from ngen.config.realization import Realization, NgenRealization
+from ngen.config.configurations import Forcing, Time
+
+#Path to directory containing this file, also contains the reouting_config.yaml template
+__conf_dir: Path = Path(__file__).parent
+
+class NgenRunException(Exception):
+    pass
+
+class RoutingRunException(Exception):
+    pass
+
+def stage_routing_config(hydrofabric, workdir: Path):
+    """prepare a routing config for testing in the provided workdir
+
+    Args:
+        workdir Path: directory to write the config to
+    """
+    with open(__conf_dir/'routing_config.yaml', 'r') as template_file:
+        template = yaml.safe_load(template_file)
+    template['run_parameters']['nts'] = 1 #FIXME
+    template['forcing_parameters']['nexus_input_folder'] = str(workdir)
+    template['output_parameters']['csv_output']['csv_output_folder'] = str(workdir)
+    template['supernetwork_parameters']['geo_file_path'] = hydrofabric
+    with open(workdir/'test_routing_config.yaml', 'w') as test_file:
+        yaml.dump(template, test_file)
+
+def make_x_walk(hydrofabric):
+    """
+
+    Args:
+        hydrofabric (_type_): _description_
+    """
+    attributes = gpd.read_file(hydrofabric, layer="flowpath_attributes").set_index('id')
```

Comment:
Not sure that this table exists anymore. Then again, I guess that is kind of the point of this utility.

Reply:
pretty sure it exists, but would be good to find out ;)